FirebasePerformance/Sources/Gauges/FPRGaugeManager.m

```diff
@@ -117,9 +117,8 @@ - (BOOL)gaugeCollectionEnabled {
 }
 
 - (void)startCollectingGauges:(FPRGauges)gauges forSessionId:(NSString *)sessionId {
-  dispatch_async(self.gaugeDataProtectionQueue, ^{
-    [self prepareAndDispatchGaugeData];
-  });
+  // Dispatch the already available gauge data with old sessionId.
+  [self prepareAndDispatchCollectedGaugeDataWithSessionId:self.currentSessionId];
```

Comment:
May it happen that the new session gauge data will be associated with the old session ID? Looks like not, because `addGaugeData` is also protected by the serial queue. But would like to check with you just in case.

Reply:
That is a good question as well. I thought about as well. The ideal solution I had in my mind was to make the prepareAndDispatch method to take both the gauge data and the session ID as inputs (making it more self-contained). But that lead to a lot of repetitive code in reseting the gauge data unprotected. So took this approach which has the risk you mentioned, but since both addGaugeData and prepareAndDispatch happen on the same serial queue, that issue would not happen.